zod: 2011-03-18 -> 2011-09-06

zeri42 opened the pull request:
- cleanup Derivation
- add map_editor and zod_launcher
- includes monkeypatch for zod_launcher windowlayout, that is incompatible with 4k monitors
- game is playable now (appears broken/orphaned in current version)


<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Mostly motivated by the game not working currently. Without the zod_launcher one has to assemble the cli-args by hand and failure modes involve ctrl+alt+f1, pkill, xrandr and are generally unpleasant. I enjoy playing this particular game a lot.

###### Things done
- version bump
 - sources still from sourceforge (unofficial github mirror is not complete)
- parallel building (-j5) (could not get -j$(nproc) to work)
- add zod_launcher binary
  - monkey patch to remove "./zod" -> "zod"
  - add scaleFactor option to resize the fixed-sized zod_launcher window
- Cleanup the Derivation
  - big let binding for various binaries that are build by 3 make calls
  - use symlinkJoin to combine the parts


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

zeri42 committed:
zod: 2011-03-18 -> 2011-09-06

- cleanup Derivation
- add map_editor and zod_launcher
- game is playable now (appears broken/orphaned in current version)
- fix comment about version
- add enableParallelBuilding=true;
- change source url to use mirror://

zeri42 commented:
ping @viric I think the original might have been you; although there is some inference here (change of username) can you please have a look?

piegamesde commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,104 @@
+    inherit name version src nativeBuildInputs buildInputs hardeningDisable NIX_LDFLAGS;
+    pname = "${name}-map_editor";
+    preBuild = "cd zod_src";
+    makeFlags = [ "map_editor" "-j6" ];
```
Same here.

piegamesde commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,104 @@
+          --replace 'message = wxT("./zod");' 'message = wxT("zod");' \
+          --replace 'SetSize(8,8,545,376);' 'SetSize(8,8,545*${toString scalingFactor},376*${toString scalingFactor});'
+        sed -i zod_launcher_src/zod_launcherFrm.cpp -e 's/wx\([PS][a-z]*\)(\([^,]*\),\([^,]*\))/wx\1(${toString scalingFactor}*\2,${toString scalingFactor}*\3)/g'
+        '';
```
Indenting (nitpick)

piegamesde commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,104 @@
+      description = "Multiplayer remake of ZED";
+      homepage = "http://zod.sourceforge.net/";
+      license = lib.licenses.gpl3Plus; /* Says the web */
+    };
```
Would you want to maintain this?

piegamesde commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,104 @@
+, scalingFactor ? 2
+}:
+let
+  version = "2011-09-06";
```
Is this really the latest version? There appears to have been some development until as late as 2018.

zeri42 replied:
it is the latest release that can be acquired from the website ... the quite recent commits are in a attached mercurial repository and I will look into bumping the version again when I had time to test them.

zeri42 committed:
maintainers: add zeri

zeri42 commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,104 @@
+, runCommandLocal
+, makeWrapper
+, coreutils
+, scalingFactor ? 2
```
well 2 is still acceptable if your screen resolution is 1024x768 or above but 1 is unreadable on 4k monitors (which while it isn't the default is still going to affect a growing number of potential users)

mweinelt commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,105 @@
+        substituteInPlace zod_launcher_src/zod_launcherFrm.cpp \
+          --replace 'message = wxT("./zod");' 'message = wxT("zod");' \
+          --replace 'SetSize(8,8,545,376);' 'SetSize(8,8,545*${toString scalingFactor},376*${toString scalingFactor});'
+        sed -i zod_launcher_src/zod_launcherFrm.cpp -e 's/wx\([PS][a-z]*\)(\([^,]*\),\([^,]*\))/wx\1(${toString scalingFactor}*\2,${toString scalingFactor}*\3)/g'
```
I think it would be much cleaner if we had this as a dedicated patch file and used `substituteAll` and passed the scalingFactor into that.

mweinelt commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,105 @@
+      # the Idea is to apply the scalingFactor to all positions and sizes and I tested 1,2,3 and 4
+      # 2,3,4 look acceptable on my 4k monitor and 1 is unreadable.
+      # also the ./ in the run command is removed to have easier time starting the game
+      patchPhase = ''
```
Don't overwrite `patchPhase`, as it (probably?) breaks `patches`, use `postPatch` instead.

SuperSandro2000 replied:
Overwriting patchPhase breaks patches.

mweinelt commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,105 @@
+      description = "Multiplayer remake of ZED";
+      homepage = "http://zod.sourceforge.net/";
+      maintainers = with maintainers; [ zeri ];
+      license = lib.licenses.gpl3Plus; /* Says the website */
```
```suggestion
      license = licenses.gpl3Plus; /* Says the website */
```

due to meta already being `with lib;`.

mweinelt replied:
Still relevant.

SuperSandro2000 commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,105 @@
+  version = "2011-09-06";
+  name = "zod-engine-${version}";
+  src = fetchzip {
+    url = "https://sourceforge.net/projects/zod/files/linux_releases/zod_linux-${version}.tar.gz";
```
Please use mirror:// style URL.

SuperSandro2000 commented on pkgs/games/zod/default.nix:
```diff
@@ -1,46 +1,105 @@
+      '';
+  };
+  zod_assets = runCommandLocal "${name}-assets" {} ''
+    mkdir -p $out/usr/lib/commander-zod
+    mkdir -p $out/usr/lib/commander-zod/blank_maps
```
```suggestion
    mkdir -p $out/usr/lib/commander-zod{,blank_maps}
```

zeri42 committed:
zod: add scaling factor

 - add patch that allows to resize the zod_launcher window at compile
 time with `scalingFactor` option
